src/NzbDrone.Core/Datastore/Migration/031_apprise_server_url.cs

```diff
@@ -0,0 +1,52 @@
+using System.Data;
+using FluentMigrator;
+using Newtonsoft.Json.Linq;
+using NzbDrone.Common.Serializer;
+using NzbDrone.Core.Datastore.Migration.Framework;
+
+namespace NzbDrone.Core.Datastore.Migration
+{
+    [Migration(031)]
+    public class apprise_server_url : NzbDroneMigrationBase
+    {
+        protected override void MainDbUpgrade()
+        {
+            Execute.WithConnection(MigrateToServerUrl);
+        }
+
+        private void MigrateToServerUrl(IDbConnection conn, IDbTransaction tran)
+        {
+            using var selectCommand = conn.CreateCommand();
+            selectCommand.Transaction = tran;
+            selectCommand.CommandText = "SELECT \"Id\", \"Settings\" FROM \"Notifications\" WHERE \"Implementation\" = 'Apprise'";
+
+            using var reader = selectCommand.ExecuteReader();
+
+            while (reader.Read())
+            {
+                var id = reader.GetInt32(0);
+                var settings = reader.GetString(1);
+
+                if (!string.IsNullOrWhiteSpace(settings))
+                {
+                    var jsonObject = Json.Deserialize<JObject>(settings);
+
+                    if (jsonObject.ContainsKey("baseUrl"))
+                    {
+                        jsonObject.Add("serverUrl", jsonObject.Value<string>("baseUrl"));
+                        jsonObject.Remove("baseUrl");
+                    }
+
+                    settings = jsonObject.ToJson();
+                }
+
+                using var updateCmd = conn.CreateCommand();
+                updateCmd.Transaction = tran;
+                updateCmd.CommandText = "UPDATE Notifications SET Settings = ? WHERE Id = ?";
```

Comment:
Previously we've had issues with `?` parameters and Postgres, can we add a test here to make sure it's working on both?

Related conversation on the Sonarr PG PR <https://github.com/Sonarr/Sonarr/pull/5559/files#r1167634345>. I had don a condition for DB Type there, but Markus believe there to be a better way for both.

Reply:
Yep, looks good!